### PR TITLE
Reorganize auth module

### PR DESCRIPTION
**Pull request by tuteng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/templates/_helpers.tpl [hunk 1]
```diff
@@ -157,4 +157,56 @@ jvmOptions:
...
+{{- end }}
+{{- join "," (compact $authenticationProviders) | quote }}
+{{- end }}
+
+{{/*
+Define function for save authenticaiton configuration
+*/}}
+{{- define "pulsar.autoConfiguration" }}
+{{- if .Values.auth.vault.enabled }}
+brokerClientAuthenticationPlugin: "org.apache.pulsar.client.impl.auth.AuthenticationToken"
```
Comment (fantapsody) [inline charts/sn-platform/templates/_helpers.tpl:184]: What if multiple auths are enabled?

File: charts/sn-platform/templates/_helpers.tpl [hunk 2]
```diff
@@ -157,4 +157,56 @@ jvmOptions:
...
+PULSAR_PREFIX_chainAuthenticationEnabled: "true"
+PULSAR_PREFIX_vaultHost: {{ template "pulsar.vault.url" . }}
+{{- if .Values.broker.readPublicKeyFromFile }}
+{{- if .Values.broker.publicKeyPath }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/_helpers.tpl:188]: It's better to use `default` instead

File: charts/sn-platform/templates/_helpers.tpl [hunk 3]
```diff
@@ -157,4 +157,56 @@ jvmOptions:
...
+{{- if .Values.auth.oauth.enabled }}
+{{- $authenticationProviders = append $authenticationProviders "io.streamnative.pulsar.broker.authentication.AuthenticationProviderOAuth" }}
+{{- end }}
+{{- join "," (compact $authenticationProviders) | quote }}
+{{- end }}
+
+{{/*
+Define function for save authenticaiton configuration
+*/}}
+{{- define "pulsar.autoConfiguration" }}
```
[the remark this answers is not in the document]
Comment (tuteng) [reply charts/sn-platform/templates/_helpers.tpl]: Yes, thanks

File: charts/sn-platform/templates/_helpers.tpl (hunk 1 above)
Comment (tuteng) [reply charts/sn-platform/templates/_helpers.tpl:184]: That will use the last brokerClientAuthenticationPlugin configuration

**Comment by tuteng** [comment]
@fantapsody PTAL

**Review by urfreespace — APPROVED** [review]

**Comment by ericsyh** [comment]
@fantapsody Can u take a review again?

**Review by maxsxu — DISMISSED** [review]
LGTM. Tested in my local env.

**Comment by ciiiii** [comment]
I found some jwt auth related configuration in values.yaml and volumeMounts definitions in templates, but no configuration in broker and proxy. 
How should we handle this, fix or remove?
https://github.com/streamnative/charts/blob/93baeacdde7800ac989fbdde32484bc85d97e4cf/charts/sn-platform/values.yaml#L352-L357 
https://cs.github.com/streamnative/charts?q=jwt%20path%3A%2F%5Echarts%5C%2Fsn-platform%5C%2F%2F

**Comment by tuteng** [comment]
Move it to next version -> sn-platform-1.6.3

**Comment by tuteng** [comment]
> I found some jwt auth related configuration in values.yaml and volumeMounts definitions in templates, but no configuration in broker and proxy. How should we handle this, fix or remove?
> 
> https://github.com/streamnative/charts/blob/93baeacdde7800ac989fbdde32484bc85d97e4cf/charts/sn-platform/values.yaml#L352-L357
> 
> 
> https://cs.github.com/streamnative/charts?q=jwt%20path%3A%2F%5Echarts%5C%2Fsn-platform%5C%2F%2F

I want to introduce support for jwt in the next pr @ciiiii

**Comment by tuteng** [comment]
@ciiiii @fantapsody PTAL I'll be adding some integration tests to the next pr, as the tests seem to bring in a lot of dependencies like hydra

**Review by maxsxu — APPROVED** [review]

File: charts/sn-platform/values.yaml [hunk 4]
```diff
@@ -385,6 +385,8 @@ auth:
     # brokerClientCredential: ""
     # brokerClientAuthenticationPlugin: org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
     # brokerClientAuthenticationParameters: ""
+    brokerClientCredentialSecret: "my-secret"
```
Comment (fantapsody) [inline charts/sn-platform/values.yaml:388]: Don't we need to comment these two values out?

Comment (ericsyh) [reply charts/sn-platform/values.yaml:388]: @tuteng PTAL

Comment (tuteng) [reply charts/sn-platform/values.yaml]: @fantapsody PTAL

**Review by maxsxu — DISMISSED** [review]

**Comment by ericsyh** [comment]
@fantapsody Pls take a review again.

**Comment by tuteng** [comment]
@fantapsody Can you take a look at this pr again? thanks

**Review by fantapsody — APPROVED** [review]

**Pull request merged**